### PR TITLE
provider/postgresql: Create provider for postgresql extensions

### PR DESCRIPTION
**Pull request by samdunne**
This change adds support for installing postgresql extensions on a
postgresql server.

### CODE REVIEW

**Comment by jen20** [comment]
cc @sean- for review

**Comment by samdunne** [comment]
@jen20 @sean- tests are fixed now

File: builtin/providers/postgresql/resource_postgresql_extension.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	query := fmt.Sprintf("CREATE EXTENSION %s", pq.QuoteIdentifier(extensionName))
+	_, err = conn.Query(query)
+	if err != nil {
+		return errwrap.Wrapf("Error creating extension: {{err}}", err)
+	}
+
+	d.SetId(extensionName)
+
+	return resourcePostgreSQLRoleRead(d, meta)
```
Comment (stack72) [inline builtin/providers/postgresql/resource_postgresql_extension.go]: Why does it call the SQLRoleRead func?

**Comment by stack72** [comment]
ping @samdunne

**Comment by samdunne** [comment]
Oops. This fell through the cracks a bit. Will rectify and commit today. Apologies @stack72

**Comment by stack72** [comment]
pinging in again LOL :)

**Comment by samdunne** [comment]
@stack72 sorry my head is like a sieve the last week. Added changes there. Pretty sure that's it

**Comment by stack72** [comment]
Hi @samdunne 

This LGTM now :) I had to make 1 small change (to fix a test) but they are green now 

```
% make testacc TEST=./builtin/providers/postgresql                                                                                                                                                      2 ↵
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/11/03 12:06:22 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/postgresql -v  -timeout 120m
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccPostgresqlDatabase_Basic
--- PASS: TestAccPostgresqlDatabase_Basic (1.49s)
=== RUN   TestAccPostgresqlDatabase_DefaultOwner
--- PASS: TestAccPostgresqlDatabase_DefaultOwner (0.53s)
=== RUN   TestAccPostgresqlExtension_Basic
--- PASS: TestAccPostgresqlExtension_Basic (0.06s)
=== RUN   TestAccPostgresqlRole_Basic
--- PASS: TestAccPostgresqlRole_Basic (0.16s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/postgresql	2.246s
```

**Comment by samdunne** [comment]
\o/ thanks @stack72

**Comment by sean-** [comment]
@samdunne Have you had a chance to poke at the extension updates that landed in 0.8?

**Comment by samdunne** [comment]
@sean- I haven't had a chance yet. Will give it a try this evening :)

**Pull request merged**